### PR TITLE
#14 Add dockcmd secrets support

### LINKED ISSUE #14: Add support for dockcmd
Add `dockcmd` support to allow for advanced secrets management for Vault and AWS Secrets Manager

### PR DESCRIPTION
**Pull request by boxboatmatt**
Signed-off-by: Matthew DeVenny <matt@boxboat.com>
Closes #14 

Add support for `dockcmd vault get-secrets` and `dockcmd aws get-secrets`.

### CODE REVIEW

File: resources/com/boxboat/jenkins/config.example.yaml [hunk 1]
```diff
@@ -42,6 +42,12 @@ vaultMap:
     secretIdCredential: vault-secret-id
     tokenCredential: vault-token
     url: http://localhost:8200
+awsProfileMap:
+  default:
+    region: us-east-1
+    accessKeyIdCredential: aws-access-key-id
```
Comment (caleblloyd) [inline resources/com/boxboat/jenkins/config.example.yaml]: should we use a user/pass credential here?  Access key ID is the User and Secret Access Key is the pass?

File: src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            Config.pipeline.sh parseAwsSecretsScript(aws.region, additionalOptions)
+        }
+    }
+
+    public parseAwsSecretsScript(String region, Map<String, Object> additionalOptions = [:]) {
+        def combinedOptions = combineOptions(options, additionalOptions)
+        return """
+            dockcmd_current_dir=\$(pwd)
+            cd "${directory}"
+            dockcmd aws get-secrets --region ${region} ${optionsString(combinedOptions)} ${files.join('" "')}
```
Comment (caleblloyd) [inline src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy]: quote "${region}"

File: src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            Config.pipeline.sh parseVaultSecretsScript(vault.url, additionalOptions)
+        }
+    }
+
+    public parseVaultSecretsScript(String vaultUrl, Map<String, Object> additionalOptions = [:]) {
+        def combinedOptions = combineOptions(options, additionalOptions)
+        return """
+            dockcmd_current_dir=\$(pwd)
+            cd "${directory}"
+            dockcmd vault get-secrets --vault-addr ${vaultUrl} ${optionsString(combinedOptions)}  ${files.join('" "')}
```
Comment (caleblloyd) [inline src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy]: quote "${vaultUrl}"

File: resources/com/boxboat/jenkins/config.example.yaml (hunk 1 above)
Comment (caleblloyd) [reply resources/com/boxboat/jenkins/config.example.yaml]: I think that for Vault we do not use user/pass currently where we could with APP_ROLE_ID and APP_ROLE_SECRET_ID.  I believe those are separate secret text credentials

Comment (boxboatmatt) [reply resources/com/boxboat/jenkins/config.example.yaml]: For AWS API authentication Access Key and Secret Key are the only authentication options (besides EC instance profile). So we will need to create a jenkins IAM user and generate access keys.

File: src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy (hunk 3 above)
Comment (boxboatmatt) [reply src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy]: Done

File: src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy (hunk 2 above)
Comment (boxboatmatt) [reply src/com/boxboat/jenkins/pipeline/common/dockcmd/DockcmdGetSecrets.groovy]: Done

File: resources/com/boxboat/jenkins/config.example.yaml (hunk 1 above)
Comment (boxboatmatt) [reply resources/com/boxboat/jenkins/config.example.yaml]: Updating naming scheme to `accessKeyIdCredential` and `secretAccessKeyCredential`
Updated `dockcmd` to use:
`${AWS_ACCESS_KEY_ID}`
`${AWS_SECRET_ACCESS_KEY}`

**Pull request merged**